patches/net/minecraft/world/item/MilkBucketItem.java.patch

```diff
@@ -1,30 +1,17 @@
 --- a/net/minecraft/world/item/MilkBucketItem.java
 +++ b/net/minecraft/world/item/MilkBucketItem.java
-@@ -22,15 +_,12 @@
+@@ -22,13 +_,10 @@
           CriteriaTriggers.CONSUME_ITEM.trigger(serverplayer, p_42923_);
           serverplayer.awardStat(Stats.ITEM_USED.get(this));
        }
 +      if (!p_42924_.isClientSide) p_42925_.curePotionEffects(p_42923_); // FORGE - move up so stack.shrink does not turn stack into air
  
        if (p_42925_ instanceof Player && !((Player)p_42925_).getAbilities().instabuild) {
           p_42923_.shrink(1);
-       }
- 
--      if (!p_42924_.isClientSide) {
--         p_42925_.removeAllEffects();
 -      }
 -
-       return p_42923_.isEmpty() ? new ItemStack(Items.BUCKET) : p_42923_;
-    }
+-      if (!p_42924_.isClientSide) {
+-         p_42925_.removeAllEffects();
```

Comment:
The fact that there is a patch change here seems somewhat weird and might be indicative of something being weird/wrong with how the patch genned. Not sure but we should look at it

Reply:
I might have fucked up a merge... this was changed in another PR

Reply:
The patch is a bit different for some reason, but essentially they are equivalent:

- this PR: <https://github.com/neoforged/NeoForge/blob/3441a232c7e8b65e3cf80ae367538d1f458e171b/patches/net/minecraft/world/item/MilkBucketItem.java.patch>
- current state of the NeoForge repo: <https://github.com/neoforged/NeoForge/blob/1.20.x/patches/net/minecraft/world/item/MilkBucketItem.java.patch>